registry/storage/vacuum.go

```diff
@@ -47,6 +53,99 @@ func (v Vacuum) RemoveBlob(dgst string) error {
 	return nil
 }
 
+// UnlinkLayer removes an entry from repository layers. withDirectory
+// allows to remove <hex-digest> directory containing the link as well.
+func (v Vacuum) UnlinkLayer(repoName string, dgst string, withDirectory bool) error {
```

Comment:
It is. The only problem is the implementation that currently removes only links. Empty directories are retained, which prolongs filesystem traversal and decreases its inodes. I'm thinking of extending a `blobStore` and/or `linkedBlobStatter` with `removeParentOnDelete bool` that would parent directory to be deleted with link itself. And it could be safely set for layer's store, `signatureStore`, and global blob store. WDYT?